apps/src/templates/manageStudents/ManageStudentsTable.jsx

```diff
@@ -0,0 +1,295 @@
+    loginType: PropTypes.string,
+  };
+
+  state = {
+    [COLUMNS.NAME]: {
+      direction: 'desc',
+      position: 0
+    }
+  };
+
+  getSortingColumns = () => {
+    return this.state.sortingColumns || {};
+  };
+
+  // The user requested a new sorting column. Adjust the state accordingly.
+  onSort = (selectedColumn) => {
+    this.setState({
+      sortingColumns: sort.byColumn({
+        sortingColumns: this.state.sortingColumns,
+        // Custom sortingOrder removes 'no-sort' from the cycle
+        sortingOrder: {
+          FIRST: 'asc',
+          asc: 'desc',
+          desc: 'asc'
+        },
+        selectedColumn
+      })
+    });
+  };
+
+  getColumns = (sortable) => {
+    const {loginType} = this.props;
+    const passwordLabel = loginType === SectionLoginType.email ? i18n.password() : i18n.secret();
+    const dataColumns = [
+      {
+        property: 'name',
+        header: {
+          label: i18n.name(),
+          props: {
+            style: {
+            ...styles.headerCell,
+            ...styles.headerCellFirst
+          }},
+          transforms: [sortable],
+        },
+        cell: {
+          format: nameFormatter,
+          props: {
+            style: {
+            ...styles.cell,
+            ...styles.cellFirst
+          }}
+        }
+      },
+      {
+        property: 'age',
+        header: {
+          label: i18n.age(),
+          props: {
+            style: {
+            ...styles.headerCell,
+          }},
+          transforms: [sortable],
+        },
+        cell: {
+          format: ageFormatter,
+          props: {
+            style: {
+            ...styles.cell,
+          }}
+        }
+      },
+      {
+        property: 'gender',
+        header: {
+          label: i18n.gender(),
+          props: {
+            style: {
+            ...styles.headerCell,
+          }},
+          transforms: [sortable],
+        },
+        cell: {
+          format: genderFormatter,
+          props: {
+            style: {
+            ...styles.cell,
+          }}
+        }
+      },
+    ];
+    const controlsColumns = [
+      {
+        property: 'password',
+        header: {
+          label: passwordLabel,
+          props: {
+            style: {
+            ...styles.headerCell,
+          }},
+        },
+        cell: {
+          format: passwordFormatter,
+          props: {
+            style: {
+            ...styles.cell,
+          }}
+        }
+      },
+      {
+        property: 'actions',
+        header: {
+          label: i18n.actions(),
+          props: {
+            style: {
+            ...styles.headerCell,
+          }},
+        },
+        cell: {
+          format: actionsFormatter,
+          props: {
+            style: {
+            ...styles.cell,
+          }}
+        }
+      },
+    ];
+
+    if (loginType === SectionLoginType.word || loginType === SectionLoginType.picture || loginType === SectionLoginType.email) {
```

Comment:
Another way you could write this if you wanted to be slightly more terse (tho no need to change):
```js
if ([SectionLoginType.word, SectionLoginType.picture, SectionLoginType.email].include(loginType)) {
```

Reply:
`.include` doesn't have IE support at all :(
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
I have done the `.indexOf` is -1 check before, but I think it looks messy

Reply:
but we should transpile/polyfill i think